### PR TITLE
Only include `labels` as a parameter if it's specified in the functio…

### PR DESCRIPTION
**Pull request by AngledLuffa**
Only include `labels` as a parameter if it's specified in the function call.  This way, encoders as opposed to classifier models can also function.  Currently only implemented for the TOKEN_CLS case

addresses https://github.com/huggingface/peft/issues/348

### CODE REVIEW

File: src/peft/peft_model.py [hunk 1]
```diff
@@ -1065,16 +1070,17 @@ def forward(
...
+            kwargs.update({
+                "input_ids": input_ids,
+                "attention_mask": attention_mask,
+                "inputs_embeds": inputs_embeds,
+                "output_attentions": output_attentions,
+                "output_hidden_states": output_hidden_states,
+                "return_dict": return_dict,
+            })
+            if labels is not None:
+                kwargs["labels"] = labels
```
Comment (pacman100) [inline src/peft/peft_model.py:1082]: Hello, I don't see the need for this, if user doesn't pass labels, it will be `None` which should work fine

**Review by pacman100 — COMMENTED** [review]
Hello @AngledLuffa, I don't see the need for the changes done in this PR. Could you share how this PR helps in solving the issues that you are facing?

File: src/peft/peft_model.py (hunk 1 above)
Comment (AngledLuffa) [reply src/peft/peft_model.py:1082]: That doesn't work if the underlying model doesn't have the `labels` parameter in its forward pass.  A model such as `roberta-large` or `vinai/phobert-large` has that problem.  I go into more details in the linked issue:  https://github.com/huggingface/peft/issues/348

Actually, I think `TOKEN_CLS` isn't exactly the right way to make use of those models.  Currently I don't believe there's any way to use peft to finetune `roberta-large` without a classifier layer, though.  Perhaps a new version such as `ENCODER` would be more appropriate.  Although in the end that would share quite a lot of code with `TOKEN_CLS`

**Comment by AngledLuffa** [comment]
> Hello @AngledLuffa, I don't see the need for the changes done in this PR. Could you share how this PR helps in solving the issues that you are facing?

The problem is that you get a `TypeError` if there is no `labels` parameter in the model's forward pass.  I go into more details in https://github.com/huggingface/peft/issues/348  Any alternative solution which allows for `roberta-large` to be used as the base model would be fine.  If you have any suggestions, I can help implement them as needed.

**Comment by AngledLuffa** [comment]
@pacman100 or others, have you thought about the difficulties in using peft for a transformer with no classification head?  It would be a useful feature to be able to use peft on the encoder / decoder / discriminator / whatever directly, allowing the user to put a custom model on top of the transformer.

Thanks!

**Comment by pacman100** [comment]
Closing this PR as per the comment on the mentioned issue https://github.com/huggingface/peft/issues/348#issuecomment-1582162135

**Comment by pacman100** [comment]
Thank you for all the work on this PR but the correct way to do would be to have a new class as you mentioned.
